nixos/modules/config/users-groups.nix

```diff
@@ -635,7 +635,7 @@ in {
           {
             assertion = let
               xor = a: b: a && !b || b && !a;
-              isEffectivelySystemUser = user.isSystemUser || (user.uid != null && user.uid < 500);
+              isEffectivelySystemUser = user.isSystemUser || (user.uid != null && user.uid < 1000);
```

Comment:
Says "between 400 and 999" above on L99 but this only checks if they're below 1000

shouldn't it check the uid is: 400 =< uid < 1000 ?

---

why are system users 400-999? what is a user below 400 then?

---

500+ used to be normal users too but most distros have moved to 1000+

Reply:
Users below 400 are also system users, but they are just static and can be found in here https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/ids.nix.

Yes, used to be normal users, the PR linked in the first message explains this was changed. :)

Reply:
:+1: ok, I think it should remain "(below X)" in the docs below so "(below 1000)", or the docs above should distinctly mention below 400 are statically reserved

Reply:
Seems to be the better option, to just keep it like `(below X)`.
Otherwise the UID description should be changed as well...

It's however weird that it's not documented/mentioned the static UIDs, I wonder how many have issues because of this.

Reply:
Resolved!